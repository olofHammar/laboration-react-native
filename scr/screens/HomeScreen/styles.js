import { StyleSheet } from "react-native";

const styles = StyleSheet.create({
  root: {
    padding: 20,
    backgroundColor: "white",
  },
  title: {
    fontSize: 18,
    fontWeight: "600",
    lineHeight: 18,
    letterSpacing: 1.01,
  },
  price: {
    fontSize: 18,
    fontWeight: "bold",
    letterSpacing: 1.01,
  },
  userInput: {
    height: 40,
    borderWidth: 1,
    borderColor: "#D1D1D1",
    paddingStart: 8,
  },
  description: {
    marginVertical: 10,
    lineHeight: 18,
    letterSpacing: 1.01,
  },
});

export default styles;
