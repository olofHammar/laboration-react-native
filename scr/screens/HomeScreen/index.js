import React, { useState, createRef } from "react";
import { View, TextInput, Text } from "react-native";
import styles from "./styles";
import welcomeData from "../../data/welcomeData";
import CustomButton from "../../components/customButton";
import ImageCarousel from "../../components/imageCarousel";
import WelcomeContainer from "../../components/welcomeContainer";
import { useGlobalContext } from "../../globalContext";

const HomeScreen = () => {
  const [input, setInput] = useState("");
  const { setCopy } = useGlobalContext();
  const inputRef = createRef();
  const [testInput, setTestInput] = useState("");

  const updateGlobalTitle = (newTitle) => {
    setCopy(newTitle);
    setTestInput(newTitle);
    updateSearchText();
  };

  const updateSearchText = () => {
    if (null !== inputRef.current) {
      inputRef.current.clear();
    }
  };

  return (
    <View style={styles.root}>
      <Text style={styles.title} testID="homeTitle">
        Memes App
      </Text>
      <Text style={{opacity: 0, height: 0}} testID="testInputText">{testInput}</Text>

      <ImageCarousel images={welcomeData.images} />

      <WelcomeContainer />

      <TextInput
        testID="userInputText"
        ref={inputRef}
        style={styles.userInput}
        placeholder="Enter what you think of memes...."
        onChangeText={(input) => setInput(input)}
        defaultValue={input}
      />

      <CustomButton
        testID={"userInputBtn"}
        text={"Update My Opinion"}
        onPress={() => updateGlobalTitle(input)}
      />
    </View>
  );
};

export default HomeScreen;
