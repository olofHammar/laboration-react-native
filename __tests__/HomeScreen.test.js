import React from "react";
import HomeScreen from "../scr/screens/HomeScreenFolder/HomeScreen";
import renderer from "react-test-renderer";
import { render, fireEvent } from "@testing-library/react-native";

it("renders correctly", () => {
  renderer.create(<HomeScreen />);
});

it("text renders correctly", () => {
  const { getByText } = render(<HomeScreen />);
  const element = getByText("Memes App");
});

it("button is clickable", () => {
  const { getByText } = render(<HomeScreen />);

  const button = getByText("Update My Opinion");
  fireEvent.press(button);
});

it("user input handles correctly", () => {
  const { getByTestId, getByText, getByPlaceholderText } = render(
    <HomeScreen />
  );

  fireEvent.changeText(
    getByPlaceholderText("Enter what you think of memes...."),
    "great"
  );

  fireEvent.press(getByText("Update My Opinion"));

  expect(getByTestId("testInputText").props.children).toBe("great");
});

it("matches snapshot", () => {
  const tree = renderer.create(<HomeScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});

