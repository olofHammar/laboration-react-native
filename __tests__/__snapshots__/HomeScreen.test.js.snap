// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "padding": 20,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 18,
        "fontWeight": "600",
        "letterSpacing": 1.01,
        "lineHeight": 18,
      }
    }
    testID="homeTitle"
  >
    Memes App
  </Text>
  <Text
    style={
      Object {
        "height": 0,
        "opacity": 0,
      }
    }
    testID="testInputText"
  >
    
  </Text>
  <View
    style={
      Object {
        "paddingVertical": 20,
      }
    }
  >
    <RCTScrollView
      data={
        Array [
          "https://hungryturtlecode.com/assets/dont_always_ie-321ee0175fd6d38744c2f2868c4eed3269777627e45c737fe0072c83c66c6b30.jpg",
          "https://bulliesout.com/wp-content/uploads/2021/03/tomemeornot3.jpg",
          "https://image.winudf.com/v2/image1/Y29tLm1uYXBwc3MuZnVubnltZW1lc19zY3JlZW5fM18xNTQ4NjUwMTk4XzA0NQ/screen-3.jpg?fakeurl=1&type=.jpg",
        ]
      }
      decelerationRate="fast"
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={true}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      showsHorizontalScrollIndicator={false}
      snapToAlignment="center"
      snapToInterval={740}
      stickyHeaderIndices={Array []}
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <View>
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
              null,
            ]
          }
        >
          <Image
            source={
              Object {
                "uri": "https://hungryturtlecode.com/assets/dont_always_ie-321ee0175fd6d38744c2f2868c4eed3269777627e45c737fe0072c83c66c6b30.jpg",
              }
            }
            style={
              Array [
                Object {
                  "height": 250,
                  "resizeMode": "contain",
                },
                Object {
                  "width": 720,
                },
              ]
            }
          />
        </View>
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
              null,
            ]
          }
        >
          <Image
            source={
              Object {
                "uri": "https://bulliesout.com/wp-content/uploads/2021/03/tomemeornot3.jpg",
              }
            }
            style={
              Array [
                Object {
                  "height": 250,
                  "resizeMode": "contain",
                },
                Object {
                  "width": 720,
                },
              ]
            }
          />
        </View>
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
              null,
            ]
          }
        >
          <Image
            source={
              Object {
                "uri": "https://image.winudf.com/v2/image1/Y29tLm1uYXBwc3MuZnVubnltZW1lc19zY3JlZW5fM18xNTQ4NjUwMTk4XzA0NQ/screen-3.jpg?fakeurl=1&type=.jpg",
              }
            }
            style={
              Array [
                Object {
                  "height": 250,
                  "resizeMode": "contain",
                },
                Object {
                  "width": 720,
                },
              ]
            }
          />
        </View>
      </View>
    </RCTScrollView>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "white",
        "margin": 10,
        "marginBottom": 20,
        "padding": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 18,
          "fontWeight": "600",
          "letterSpacing": 1.01,
          "lineHeight": 18,
        }
      }
    >
      Memes
    </Text>
    <Text
      style={
        Object {
          "letterSpacing": 1.01,
          "lineHeight": 18,
          "marginVertical": 10,
        }
      }
    >
      An Internet meme, more commonly known simply as a meme (/miÀêm/ MEEM), is an idea, behavior, or style (meme) that is spread via the Internet, often through social media platforms and especially for humorous purposes. What is considered a meme may vary across different communities on the Internet and is subject to change over time.
    </Text>
    <Text
      style={
        Object {
          "fontSize": 15,
          "fontWeight": "600",
          "letterSpacing": 1.01,
          "lineHeight": 18,
          "marginTop": 10,
        }
      }
    >
      I think memes are 
      
    </Text>
  </View>
  <TextInput
    allowFontScaling={true}
    defaultValue=""
    onChangeText={[Function]}
    placeholder="Enter what you think of memes...."
    rejectResponderTermination={true}
    style={
      Object {
        "borderColor": "#D1D1D1",
        "borderWidth": 1,
        "height": 40,
        "paddingStart": 8,
      }
    }
    testID="userInputText"
    underlineColorAndroid="transparent"
  />
  <View
    accessible={true}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#e47911",
          "borderColor": "#a15e1b",
          "borderRadius": 5,
          "borderWidth": 1.5,
          "height": 45,
          "justifyContent": "center",
          "marginVertical": 10,
        },
        undefined,
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontSize": 16,
            "fontWeight": "400",
          },
          undefined,
        ]
      }
    >
      Update My Opinion
    </Text>
  </View>
</View>
`;
